050-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,744 @@
+* On attempts to change a `KafkaTopic`'s `spec.topicName`.
+  For UTO this will be reported via the status only.
+* On detection of conflicting changes between Kube and Kafka.
+  For UTO such conflicts cannot happen.
+* On attempts to decrease the number of partitions.
+  For UTO this will be reported via the status only.
+
+Thus, the UTO will emit no events.
+Any software which consumed BTO events will not be compatible with UTO.
+
+## Migration
+
+### Standalone case
+
+Existing users of the standalone BTO would have to:
+
+1. Review how they're using the BTO and whether they require bidirectionality
+   1. If bidirectional support is required by their usage then the UTO cannot be used. The user will have no way of managing topics using `KafkaTopic` resources once ZooKeeper support is removed by Apache Kafka and/or Strimzi.
+2. Undeploy the BTO (they can retain their existing `KafkaTopics`)
+3. Deploy the UTO (reconfiguring their `Kafka` CR, or raw `Deployment` in the standalone case). 
+4. Some reconfiguration of pod resources and JVM options may also be required. 
+
+### CO-Managed case
+
+A new feature gate will be introduced to allow users to adopt the UTO at their own pace.
+The new Feature gate will be named `UnidirectionalTopicOperator`.
+The following table shows the expected graduation of the `UnidirectionalTopicOperator` feature gate:
+
+| Phase | Strimzi versions       | Default state                                          |
+|:------|:-----------------------|:-------------------------------------------------------|
+| Alpha | 0.35 - 0.37            | Disabled by default                                    |
+| Beta  | 0.38 - 0.39            | Enabled by default                                     |
+| GA    | 0.40 and newer         | Enabled by default (without possibility to disable it) |
+
+### Disadvantages
+
+#### Use of finalizers
+
+Using finalizers on `KafkaTopics` means that the operator has to be running (to remove the finalizer) before the resources will actually be deleted.
+That can prevent other resources from being removed (for example the `Namespace` containing the `KafkaTopics`).
+In these cases it is possible to remove the finalizer manually (e.g. via `kubectl edit`), which will then allow the apiserver to remove any other resources.
+
+#### Compatibility with other tooling
+
+This approach is not easily compatible with other topic management tooling.
+This includes things like the `kafka-configs.sh` and other Apache Kafka scripts as well as any other tooling which uses the `Admin` client, including Kafka applications themselves.
+
+In the particular case of a hypthetical "Admin server" that was providing CLI or UI functionality in a Strimzi-aware way, it _might_ be possible to use wrap `Admin` (decorator pattern), or write it using some abstraction over `Admin` which could also support operations on `KafkaTopics` for those interactions which would otherwise result in conflicting sources of truth.
```

Comment:
```suggestion
In the particular case of a hypothetical "Admin server" that was providing CLI or UI functionality in a Strimzi-aware way, it _might_ be possible to use wrap `Admin` (decorator pattern), or write it using some abstraction over `Admin` which could also support operations on `KafkaTopics` for those interactions which would otherwise result in conflicting sources of truth.
```